src/ruleset.ts

```diff
@@ -278,5 +278,20 @@ export default {
       formats: [oas3],
       severity: DiagnosticSeverity.Warning,
     },
+    // Author: Advanced API & Integrations Team (https://www.oneadvanced.com/)
+    "no-file-extensions-in-paths": {
+      description: "Paths must not include file extensions such as .json, .xml, .html and .txt",
+      message:
+        "Paths must not include file extensions such as .json, .xml, .html and .txt. Use the OpenAPI `content` keyword to tell consumers which Media Types are available.",
+        given: "$.paths[*]~",
+        then: {
+        function: pattern,
+        functionOptions: {
+          notMatch: ".(\\.json|\\.xml|\\.html|\\.txt)$",
+        },
+      },
+      formats: [oas3],
```

Comment:
Any reason this is only being applied to oas3? I think this will work on OAS2 as well.

Reply:
Oversight is the only reason. Let me give it some tests and if it's fine, add it there too.

Reply:
@savage-alex can you see if this works on oas2 now we've fixed #32.

Reply:
```suggestion
```